File: test-data/perfect_match_test.ts
// Test schemas that should be perfectly equivalent cross-language

// [agree:perfect:zod]
export const PerfectSchema = z.object({
  id: z.number(),        // Should match int/integer
  name: z.string(),      // Should match str/string
  active: z.boolean(),   // Should match bool/boolean
  score: z.number(),     // Should match float (both map to number)
  email: z.string().email(), // Should match EmailStr (both map to email)
});
export type Perfect = z.infer<typeof PerfectSchema>;
// [agree:end]

File: test-data/perfect_match_test.py
# Test schemas that should be perfectly equivalent cross-language

# [agree:perfect:pydantic]
class PerfectSchema(BaseModel):
    id: int
    name: str
    active: bool
    score: float
    email: EmailStr
# [agree:end]

# [agree:perfect:sqlalchemy]
class Perfect(Base):
    __tablename__ = "perfect"
    id = Column(Integer, primary_key=True)
    name = Column(String, nullable=False)
    active = Column(Boolean, default=True)
    score = Column(Float)
    email = Column(String)  # Will map to string, not email
# [agree:end]